src/main/java/words/Words.java

```diff
@@ -0,0 +1,17 @@
+package words;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Words {
+    public static String matches(String regex, String text) {
```

Comment:
It would also be nice to have an instance method instead of class one. Like that:

```
public class Words {
  private static final PATTERN = Pattern.compile("%REGEX%");
  public List<String> getWords(String value){
    //implementation
  }
}
```
and tests like that:
```
private final Words words = new Words();
@Test
void singleWord(){
  assertThat(words.getWords("Some")).equalsTo(List.of("Some"));
}

@Test
void wordsStartingWithLowerCase_Ignored(){
  assertThat(words.getWords("lower Some lower")).equalsTo(List.of("Some"));
}

@Test
void wordsSingleLetter(){
  assertThat(words.getWords("A lower Some lower")).equalsTo(List.of("A", "Some"));
}

@Test
void task(){
  assertThat(words.getWords("Word cAT DATA A site Java tExt")).equalsTo(List.of("Word", "DATA", "A", "Java"));
}
```

Reply:
updated